system/src/commonMain/kotlin/com/splendo/kaluga/system/network/services/NetworkManagerService.kt

```diff
@@ -0,0 +1,32 @@
+/*
+ Copyright 2020 Splendo Consulting B.V. The Netherlands
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+ */
+
+package com.splendo.kaluga.system.network.services
+
+import com.splendo.kaluga.system.network.Network
+
+interface NetworkManagerHandler {
+    fun startNotifier()
+    fun stopNotifier()
+}
+
+interface NetworkManagerService {
+    fun setIsNetworkEnabled(newValue: Boolean)
+    fun getIsNetworkEnabled(): Boolean
```

Comment:
We have properties for that, but even if not using properties the syntax is wrong (`getIs`)

However, we already have a sealed class that tells us the `set` state (`handleStateChange`), this, they can just have a boolean property or have a subclass `Enabled` and `Disabled` that they inherit from.

`NetworkManagerService` is also a confusing name, you can just replace it with `(Network)->()`,  `getIsNetworkEnabled` is also not used as far as I can see, and would be rather pointless.

Reply:
Sorry I didn't understand. What do you mean?

> NetworkManagerService is also a confusing name, you can just replace it with (Network)->()

Reply:
This interfact has only a single method, and you only use it for callbacks.

You can remove it, and replace it with a `callback: (Network)->()` parameter.

Reply:
Thanks, after reading more carefully what you said here, I realized what you meant and the work for this change is in progress.